docs/src/chainrules.md

```diff
@@ -0,0 +1,243 @@
+    
+    # Ramp rates
+    @constraint(model, [g in unit_codes, t in 2:n_periods], p[g, t] - p[g, t - 1] <= 60 * RR[g])
+    @constraint(model, [g in unit_codes], p[g, 1] - P0[g] <= 60 * RR[g])
+    @constraint(model, [g in unit_codes, t in 2:n_periods], p[g, t - 1] - p[g, t] <= 60 * RR[g])
+    @constraint(model, [g in unit_codes], P0[g] - p[g, 1] <= 60 * RR[g])
+    
+    # Objective
+    @objective(
+        model,
+        Min,
+        sum((Cp[g] * p[g, t]) + (Cnl[g] * u[g, t]) for g in unit_codes, t in 1:n_periods),
+    )
+    
+    optimize!(model)
+    # asserting finite optimal value
+    @assert termination_status(model) == MOI.OPTIMAL
+    # converting to dense matrix
+    return JuMP.value.(p.data)
+end
+```
+
+```@example 1
+m = Model(Clp.Optimizer)
+@show unit_commitment(
+    [1.0, 1.2, 1.4, 1.6], [1.0, 1.2, 1.4, 1.6],
+    [1000.0, 1500.0], [500.0, 1000.0],
+    model=m, silent=true
+)
+println(m)
+```
+
+```@example 1
+demand_values = 0.05:0.05:3.0
+pvalues = [unit_commitment([1.0, di, 1.4, 1.6], [1.0, di, 1.4, 1.6], [1000.0, 1500.0], [500.0, 1000.0], silent=true) for di in demand_values]
+pflat = [getindex.(pvalues, i) for i in eachindex(pvalues[1])]
+```
+
+```@example 1
+scatter(demand_values, pflat, size=(1500,1000), legendfontsize=20, markersize=20)
+title!("Generation at different time frames and generators for a single variation")
+```
+
+## Forward Differentiation
+
+Forward differentiation rule for the solution map of the unit commitment problem.
+It takes as arguments:
+1. the perturbations on the input parameters
+2. the differentiated function
+3. the primal values of the input parameters,
+
+and returns a tuple `(primal_output, perturbations)` propagated to the result:
+
+```@example 1
+function ChainRulesCore.frule(
+        (_, Δload1_demand, Δload2_demand, Δgen_costs, Δnoload_costs),
+        ::typeof(unit_commitment),
+        load1_demand, load2_demand, gen_costs, noload_costs;
+        optimizer=Clp.Optimizer,
+        )
+    # creating the UC model with a DiffOpt optimizer wrapper around Clp
+    model = Model(() -> diff_optimizer(optimizer))
+    # building and solving the main model
+    pv = unit_commitment(load1_demand, load2_demand, gen_costs, noload_costs, model=model)
+    energy_balance_cons = model[:energy_balance_cons]
+
+    # Setting some perturbation of the right-hand side of the energy balance constraints
+    # the RHS is equal to the sum of load demands at each period.
+    # the corresponding perturbation are set accordingly as the set of perturbations of the two loads
+    MOI.set.(
+        model,
+        DiffOpt.ForwardInConstraint(), energy_balance_cons,
+        [d1 + d2 for (d1, d2) in zip(Δload1_demand, Δload2_demand)],
+    )
+
+    p = model[:p]
+    u = model[:u]
+
+    # setting the perturbation of the linear objective
+    for t in size(p, 2)
+        MOI.set(model, DiffOpt.ForwardInObjective(), p[1,t], Δgen_costs[1])
```

Comment:
These should be updated, ForwardInConstraint and ForwardInObjective now takes the function as argument.
See
https://github.com/jump-dev/DiffOpt.jl/blob/2a373d07b1c45b02b5d03c9699ee6b23f70297bc/examples/chainrules.jl#L93
https://github.com/jump-dev/DiffOpt.jl/blob/2a373d07b1c45b02b5d03c9699ee6b23f70297bc/examples/chainrules.jl#L100-L101